eMPRess_GUI_version1.py

```diff
@@ -0,0 +1,354 @@
+        #trans_entry.bind('<FocusOut>', self.check_trans_input)  # same as above 
+        #loss_entry.bind('<FocusOut>', self.check_loss_input)  # same as above 
+        dup_entry.grid(row=0, column=1, sticky="w")
+        trans_entry.grid(row=1, column=1, sticky="w")
+        loss_entry.grid(row=2, column=1, sticky="w")
+
+    def plot_cost_landscape(self):
+        """Plots the cost landscape using matplotlib and embeds the graph in a tkinter window."""    
+        # creates a new tkinter window 
+        plt_window = tk.Toplevel(self.master)
+        plt_window.geometry("550x550")
+        plt_window.title("Matplotlib Graph DEMO")
+        # creates a new frame
+        plt_frame = tk.Frame(plt_window)
+        plt_frame.pack(fill=tk.BOTH, expand=1)
+        plt_frame.pack_propagate(False)
+        # use empress wrapper
+        recon_input = empress.read_input(self.file_path)
+        cost_regions = empress.compute_cost_regions(recon_input, 0.5, 10, 0.5, 10)  
+        #cost_regions.draw_to_file('./examples/cost_poly.png')  # draw and save to a file
+        fig = cost_regions.draw()  # draw to figure (creates matplotlib figure)
+        canvas = FigureCanvasTkAgg(fig, plt_frame)
+        canvas.draw()
+        canvas.get_tk_widget().pack(side=tk.TOP, fill=tk.BOTH, expand=True)
+        toolbar = NavigationToolbar2Tk(canvas, plt_frame)
+        toolbar.update()
+        canvas.get_tk_widget().pack(side=tk.TOP)
+        # updates the DTL costs using the x,y coordinates clicked by the user inside the graph 
+        # otherwise pops up a warning message window
+        fig.canvas.callbacks.connect('button_press_event', self.get_xy_coordinates)
+
+    def get_xy_coordinates(self, event):
+        """Updates the DTL costs when user clicks on the matplotlib graph, otherwise pops up a warning message window."""
+        if event.inaxes is not None:
+            self.dup_cost.set("1")
+            self.trans_cost.set(str(round(event.ydata, 2)))
+            self.loss_cost.set(str(round(event.xdata, 2)))
+            # enables the next step, viewing reconciliation result
+            self.has_dup_cost = True
+            self.has_trans_cost = True
+            self.has_loss_cost = True
+            self.enable_recon_btn()
+        else:
+            messagebox.showinfo("Warning", "Please click inside the axes bounds.")
+
+    def enable_recon_btn(self):
+        """Enables the next step, viewing reconciliation results, after DTL costs are set correctly."""
+        if self.has_dup_cost == True and self.has_trans_cost == True and self.has_loss_cost == True:
+            self.compute_recon_button.configure(state=tk.NORMAL)
+
+    def check_dup_input(self, *args):
+        """Checks and updates the duplication cost when user enters a value in the entry box."""
+        dupValue_without_decimal_point = self.dup_cost.get().replace(".", "")
```

Comment:
I'm not sure what you are doing here. If you want to round the decimal off, convert it to `int`, if you want to round to the nearest 2nd decimal place, see https://docs.python.org/3/library/functions.html#round and https://www.tutorialspoint.com/python3/number_round.htm . The code should look something like this:
```python
x = "1.732"
y = float(x)
z = round(y, 2) # now y = 1.73
if z <= 0:
    print("the number is negative!")
```

Reply:
I wanted to use str.isdigit() to check if the input contains only numbers instead of letters, but this function treats the decimal point as invalid input, so I removed the decimal point if it was there before using str.isdigit().